### ISSUE TITLE
TypeError in testing MAC bits

### ISSUE THREAD

**Issue by shpedoikal** [portcullislabs/sslxray#1]
Hi Graham, this is a run against `openssl s_server -accept 127.0.0.1:8888 -cert rsa.cert.pem` as a test. I can share the rsa key/cert if need be for testing. `OpenSSL 1.1.0g  2 Nov 2017`. `Python 2.7.15rc1` on Ubuntu 18.04.

```
python sslxray.py -p 8888 -s 127.0.0.1
Scanning 127.0.0.1 (127.0.0.1:8888).  [SNI: None]
Enumerating ciphers...
Testing protocol SSLv2.0...
Testing protocol SSLv3.0...
Testing protocol TLSv1.0...
	RSA_WITH_AES_128_CBC_SHA [0x0000002f]
	DHE_RSA_WITH_AES_128_CBC_SHA [0x00000033]
	RSA_WITH_AES_256_CBC_SHA [0x00000035]
	DHE_RSA_WITH_AES_256_CBC_SHA [0x00000039]
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013]
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014]
Testing protocol TLSv1.1...
	RSA_WITH_AES_128_CBC_SHA [0x0000002f]
	DHE_RSA_WITH_AES_128_CBC_SHA [0x00000033]
	RSA_WITH_AES_256_CBC_SHA [0x00000035]
	DHE_RSA_WITH_AES_256_CBC_SHA [0x00000039]
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013]
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014]
Testing protocol TLSv1.2...
	RSA_WITH_AES_128_CBC_SHA [0x0000002f]
	DHE_RSA_WITH_AES_128_CBC_SHA [0x00000033]
	RSA_WITH_AES_256_CBC_SHA [0x00000035]
	DHE_RSA_WITH_AES_256_CBC_SHA [0x00000039]
	RSA_WITH_AES_128_CBC_SHA256 [0x0000003c]
	RSA_WITH_AES_256_CBC_SHA256 [0x0000003d]
	DHE_RSA_WITH_AES_128_CBC_SHA256 [0x00000067]
	DHE_RSA_WITH_AES_256_CBC_SHA256 [0x0000006b]
	RSA_WITH_AES_128_GCM_SHA256 [0x0000009c]
	RSA_WITH_AES_256_GCM_SHA384 [0x0000009d]
	DHE_RSA_WITH_AES_128_GCM_SHA256 [0x0000009e]
	DHE_RSA_WITH_AES_256_GCM_SHA384 [0x0000009f]
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013]
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014]
	ECDHE_RSA_WITH_AES_128_CBC_SHA256 [0x0000c027]
	ECDHE_RSA_WITH_AES_256_CBC_SHA384 [0x0000c028]
	ECDHE_RSA_WITH_AES_128_GCM_SHA256 [0x0000c02f]
	ECDHE_RSA_WITH_AES_256_GCM_SHA384 [0x0000c030]
Testing protocol TLSv1.3...
Testing elliptic curve support...
Testing elliptic curve support for ECDHE_RSA_WITH_AES_128_CBC_SHA ...
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_256_CBC_SHA ...
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_128_CBC_SHA ...
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_256_CBC_SHA ...
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_128_CBC_SHA ...
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA [0x0000c013] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_256_CBC_SHA ...
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA [0x0000c014] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_128_CBC_SHA256 ...
	ECDHE_RSA_WITH_AES_128_CBC_SHA256 [0x0000c027] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA256 [0x0000c027] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_128_CBC_SHA256 [0x0000c027] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_256_CBC_SHA384 ...
	ECDHE_RSA_WITH_AES_256_CBC_SHA384 [0x0000c028] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA384 [0x0000c028] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_256_CBC_SHA384 [0x0000c028] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_128_GCM_SHA256 ...
	ECDHE_RSA_WITH_AES_128_GCM_SHA256 [0x0000c02f] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_128_GCM_SHA256 [0x0000c02f] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_128_GCM_SHA256 [0x0000c02f] curve: [0x0019] secp521r1
Testing elliptic curve support for ECDHE_RSA_WITH_AES_256_GCM_SHA384 ...
	ECDHE_RSA_WITH_AES_256_GCM_SHA384 [0x0000c030] curve: [0x0017] secp256r1
	ECDHE_RSA_WITH_AES_256_GCM_SHA384 [0x0000c030] curve: [0x0018] secp384r1
	ECDHE_RSA_WITH_AES_256_GCM_SHA384 [0x0000c030] curve: [0x0019] secp521r1
Server did not advertise NPN support.
The server supports the following outdated ALPN protocols: h2-13, h2-07, h2-08, h2-01, h2-12, h2-08, h2-02, h2-06, h2-03, h2-11, h2-04, h2-05, h2-10
Server advertised ALPN support.
Testing validation of individual MAC bits...
+Traceback (most recent call last):
  File "sslxray.py", line 419, in <module>
    testMACValidation()
  File "sslxray.py", line 275, in testMACValidation
    tls.handshakeClientCert()
  File "/root/sslxray.git/tlslite/tlsconnection.py", line 307, in handshakeClientCert
    for result in handshaker:
  File "/root/sslxray.git/tlslite/tlsconnection.py", line 323, in _handshakeClientAsync
    for result in self._handshakeWrapperAsync(handshaker, checker):
  File "/root/sslxray.git/tlslite/tlsconnection.py", line 1724, in _handshakeWrapperAsync
    for result in handshaker:
  File "/root/sslxray.git/tlslite/tlsconnection.py", line 471, in _handshakeClientAsyncHelper
    serverHello.tackExt):
  File "/root/sslxray.git/tlslite/tlsconnection.py", line 776, in _clientRSAKeyExchange
    HandshakeType.certificate, certificateType):
  File "/root/sslxray.git/tlslite/tlsrecordlayer.py", line 802, in _getMsg
    yield Certificate(constructorType).parse(p)
  File "/root/sslxray.git/tlslite/messages.py", line 474, in parse
    x509.parseBinary(certBytes)
  File "/root/sslxray.git/tlslite/x509.py", line 97, in parseBinary
    self.publicKey = _createPublicRSAKey(n, e)
  File "/root/sslxray.git/tlslite/utils/keyfactory.py", line 179, in _createPublicRSAKey
    return PyCrypto_RSAKey(n, e)
  File "/root/sslxray.git/tlslite/utils/pycrypto_rsakey.py", line 18, in __init__
    self.rsa = RSA.construct( (n, e) )
  File "/usr/lib/python2.7/dist-packages/Crypto/PublicKey/RSA.py", line 539, in construct
    key = self._math.rsa_construct(*tup)
TypeError: argument 2 must be long, not int
```

**Comment by gsuberland** [comment]
Apologies, I did fix this but forgot to push it to GitHub! I'll have a once-over of the code and get it sorted.

**Comment by gsuberland** [comment]
This should be fixed in f2ee922. Feel free to re-open if I broke something!
